sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/model/PartitionMetadata.java

```diff
@@ -28,13 +30,31 @@
 import org.apache.beam.sdk.coders.DefaultCoder;
 import org.apache.beam.sdk.io.gcp.spanner.changestreams.encoder.TimestampEncoding;
 import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Sets;
 
 /** Model for the partition metadata database table used in the Connector. */
 @SuppressWarnings("initialization.fields.uninitialized") // Avro requires the default constructor
 @DefaultCoder(AvroCoder.class)
 public class PartitionMetadata implements Serializable {
 
   private static final long serialVersionUID = 995720273301116075L;
+  /**
+   * We use a bogus partition here to estimate the average size of a partition metadata record.
+   *
+   * <p>The only dynamically allocated size field here is the "parentTokens", which is a set and can
+   * expand. In practice, however, partitions have 1 to 2 parents at most.
+   */
+  public static final long AVERAGE_PARTITION_BYTES_SIZE =
+      PartitionMetadata.newBuilder()
+          .setPartitionToken(InitialPartition.PARTITION_TOKEN)
+          .setParentTokens(Sets.newHashSet("fake_parent"))
```

Comment:
can the parent token string be large? does "fake_parent" estimate it well?

Reply:
Good point, I created a sample token which is more realistic.